src/core/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,166 @@
+ *   <BreadcrumbItem onClick={() => {}}>Item 2</BreadcrumbItem>
+ * </Breadcrumbs>
+ />
+ */
+export const Breadcrumbs = React.forwardRef(
+  (props: BreadcrumbsProps, ref: React.RefObject<HTMLElement>) => {
+    const {
+      children,
+      currentIndex = React.Children.count(children) - 1,
+      separator,
+      className,
+      ...rest
+    } = props;
+    const items = React.Children.toArray(children);
+
+    useTheme();
+
+    const breadcrumbsRef = React.useRef<HTMLElement>(null);
+
+    const [visibleCount, setVisibleCount] = React.useState(items.length);
+    const overflowBreakpoints = React.useRef<number[]>([]);
+
+    React.useLayoutEffect(() => {
+      setVisibleCount(React.Children.count(children));
+      overflowBreakpoints.current = [];
+    }, [children]);
+
+    const [breadcrumbsWidth, setBreadcrumbsWith] = React.useState<number>(0);
+    const updateBreadcrumbsWidth = React.useCallback(
+      ({ width }) => setBreadcrumbsWith(width),
+      [],
+    );
+    const [resizeRef] = useResizeObserver(updateBreadcrumbsWidth);
+    const refs = useMergedRefs(breadcrumbsRef, resizeRef, ref);
+
+    React.useLayoutEffect(() => {
+      if (!breadcrumbsRef.current) {
+        return;
+      }
+
+      // hide items when there's no space available
+      if (
+        breadcrumbsRef.current.offsetWidth <
+          breadcrumbsRef.current.scrollWidth &&
+        visibleCount > 1
+      ) {
+        setVisibleCount((count) => count - 1);
+        overflowBreakpoints.current.push(breadcrumbsRef.current.offsetWidth);
+      }
+      // restore items when there's enough space again
+      else if (
+        overflowBreakpoints.current.length > 0 &&
+        breadcrumbsRef.current.offsetWidth >
+          overflowBreakpoints.current[overflowBreakpoints.current.length - 1]
+      ) {
+        setVisibleCount((count) => count + 1);
+        overflowBreakpoints.current.pop();
+      }
+    }, [breadcrumbsWidth, visibleCount]);
+
+    const Separator = () => (
+      <li className='iui-breadcrumbs-separator' aria-hidden>
+        {separator ?? <SvgChevronRight />}
+      </li>
+    );
+
+    const ListItem = ({ index }: { index: number }) => {
+      const item = items[index];
+      return (
+        <li
+          className={cx('iui-breadcrumbs-item', {
+            'iui-current': currentIndex === index,
+          })}
+        >
+          {React.isValidElement(item) &&
+            React.cloneElement(item, {
+              'aria-current':
+                item.props['aria-current'] ?? currentIndex === index
+                  ? 'location'
+                  : undefined,
+            })}
+        </li>
+      );
+    };
+
+    return (
+      <nav
+        className={cx('iui-breadcrumbs', className)}
+        ref={refs}
+        aria-label='Breadcrumb'
+        {...rest}
+      >
+        <ol className='iui-breadcrumbs-list'>
+          {visibleCount > 1 && <ListItem index={0} />}
+          {visibleCount > 1 && <Separator />}
```

Comment:
Maybe wrap with fragment and put item with separator under same condition.

Reply:
Done